src/sentry/notifications/types.py

```diff
@@ -70,21 +70,38 @@ class NotificationSettingTypes(ValueEqualityEnum):
     SPIKE_PROTECTION = 60
 
 
+class NotificationSettingEnum(Enum):
+    DEFAULT = "default"
+    DEPLOY = "deploy"
+    ISSUE_ALERTS = "alerts"
+    WORKFLOW = "workflow"
+    ACTIVE_RELEASE = "activeRelease"
+    APPROVAL = "approval"
+    QUOTA = "quota"
+    QUOTA_ERRORS = "quotaErrors"
+    QUOTA_TRANSACTIONS = "quotaTransactions"
+    QUOTA_ATTACHMENTS = "quotaAttachments"
+    QUOTA_REPLAYS = "quotaReplays"
+    QUOTA_WARNINGS = "quotaWarnings"
+    QUOTA_SPEND_ALLOCATIONS = "quotaSpendAllocations"
+    SPIKE_PROTECTION = "spikeProtection"
+
+
 NOTIFICATION_SETTING_TYPES = {
-    NotificationSettingTypes.DEFAULT: "default",
-    NotificationSettingTypes.DEPLOY: "deploy",
-    NotificationSettingTypes.ISSUE_ALERTS: "alerts",
-    NotificationSettingTypes.WORKFLOW: "workflow",
-    NotificationSettingTypes.ACTIVE_RELEASE: "activeRelease",
-    NotificationSettingTypes.APPROVAL: "approval",
-    NotificationSettingTypes.QUOTA: "quota",
-    NotificationSettingTypes.QUOTA_ERRORS: "quotaErrors",
-    NotificationSettingTypes.QUOTA_TRANSACTIONS: "quotaTransactions",
-    NotificationSettingTypes.QUOTA_ATTACHMENTS: "quotaAttachments",
-    NotificationSettingTypes.QUOTA_REPLAYS: "quotaReplays",
-    NotificationSettingTypes.QUOTA_WARNINGS: "quotaWarnings",
-    NotificationSettingTypes.QUOTA_SPEND_ALLOCATIONS: "quotaSpendAllocations",
-    NotificationSettingTypes.SPIKE_PROTECTION: "spikeProtection",
+    NotificationSettingTypes.DEFAULT: NotificationSettingEnum.DEFAULT.value,
```

Comment:
i'm really confused why we have to map enum values to enum values everywhere here. it seems like this will be annoying to maintain. any way we could get around not doing this?

Reply:
@JoshFerge this is temporary until we finish the migration. It's because in the old models we need int enums and the new model we need string enums.

Reply:
can you add a TODO comment explaining this? thanks